docs/TreeInput.md

```diff
@@ -0,0 +1,189 @@
+  <source src="./img/ReferenceNodeInput-TreeInput-multiple.webm" type="video/webm"/>
+  <source src="./img/ReferenceNodeInput-TreeInput-multiple.mp4" type="video/mp4"/>
+  Your browser does not support the video tag.
+</video>
+
+
+```tsx
+import { SimpleForm } from 'react-admin';
+import { TreeInput } from '@react-admin/ra-tree';
+import treeData from './treeData';
+
+export const SimpleTreeForm = () => (
+    <SimpleForm>
+        <TreeInput source="category" treeData={treeData} multiple />
+    </SimpleForm>
+);
+```
+
+## `titleField`
+
+Use the `titleField` prop to specify the name of the field holding the node title:
+
+```tsx
+<TreeInput 
+    source="category" 
+    treeData={treeData} 
+    titleField="name" 
+/>
+```
+
+## `treeData`
+
+The list of possible choices must be passed as the `treeData` prop. It must be an array of nodes with a `children` field.
+
+```tsx
+<TreeInput source="category" treeData={[
+    { id: 1, title: 'Clothing', isRoot: true, children: [2, 6] },
+    { id: 2, title: 'Men', children: [3] },
+    { id: 3, title: 'Suits', children: [4, 5] },
+    { id: 4, title: 'Slacks', children: [] },
+    { id: 5, title: 'Jackets', children: [] },
+    { id: 6, title: 'Women', children: [7, 10, 11] },
+    { id: 7, title: 'Dresses', children: [8, 9] },
+    { id: 8, title: 'Evening Gowns', children: [] },
+    { id: 9, title: 'Sun Dresses', children: [] },
+    { id: 10, title: 'Skirts', children: [] },
+    { id: 11, title: 'Blouses', children: [] },
+]} />
+```
+
+If you need to fetch the `treeData`, you're probably editing a relationship. In that case, you should use the [`<ReferenceNodeInput>`](./ReferenceNodeInput.md) component, which fetches the `treeData` from a reference resource on mount .
+
+## Fetching Choices
+
+You can use `dataProvider.getTree()` to fetch choices. For example, to fetch a list of categories for a product:
+
+```tsx
+import { useGetTree, TreeInput } from '@react-admin/ra-tree';
+
+const CategoryInput = () => {
+    const { isLoading, data: tree } = useGetTree('categories');
+    if (isLoading) return <Loading />;
+    return (
+        <TreeInput 
+            source="category" 
+            treeData={tree} 
+        />
+    );
+};
+```
+
+The `isLoading` prop is used to display a loading indicator while the data is being fetched.
+
+However, most of the time, if you need to populate a `<TreeInput>` with choices fetched from another resource, it's because you are trying to set a foreign key. In that case, you should use [`<ReferenceNodeInput>`](./ReferenceNodeInput.md) to fetch the choices instead (see next section). 
+
+## Selecting a Foreign Key
+
+If you use `<TreeInput>` to set a foreign key for a many-to-one or a one-to-one relationship, you’ll have to [fetch choices](#fetching-choices), as explained in the previous section. You’ll also have to fetch the record corresponding to the current value of the foreign key, as it may not be in the list of choices.
```

Comment:
> You’ll also have to fetch the record corresponding to the current value of the foreign key, as it may not be in the list of choices.

This may be misleading as this is not implemented in `ReferenceNodeInput` yet